components/header/TypeSuggestionsBox.tsx

```diff
@@ -0,0 +1,63 @@
+import { View, Text, FlatList, TouchableOpacity } from "react-native";
+import { createStyleSheet, useStyles } from "react-native-unistyles";
+import { GeoNamesData } from "../../types/geoNamesSchema";
+import { searchBarContainerHeight } from "../../constants";
+
+interface TypeSuggestionsBoxProps {
+  data: GeoNamesData;
```

Comment:
Można by się pokusić o to, żeby ten komponent nie był ściśle związany z `GeoNamesData`.  Zamiast tego on mógłby przyjmować tylko to, czego faktycznie potrzebuje do wyswietlenia - obiektów, które mają jakiś tekst i jakieś id (do keyExtractora)